base/multi.jl

```diff
@@ -980,6 +980,7 @@ function start_worker(out::IO)
     print(out, LPROC.bind_addr)
     print(out, '\n')
     flush(out)
+    ccall((:sync, "libc"), Void, ())   # necessary for SGE, which redirects and buffers stdout
```

Comment:
The whole point is that flush() doesn't do this in this case.  

In the case of addprocs_sge(), out == STDOUT.  The worker flushes STDOUT, but apparently SGE redirects stdout to a file locally at the worker, and somehow adds its own buffering again in that process.  This makes the flush() useless, but the sync() seems to circumvent the problem, because STDOUT has turned into a file on the filesystem.